### ISSUE TITLE
Automatically redirect VS is created on all vlans despite https VS have vlans specified.

### ISSUE THREAD

**Issue by LukaszBrzoskof5** [F5Networks/f5-appsvcs-extension#514]
### Environment
 * Application Services Version: 3.29.0
 * BIG-IP Version: 16.1.0

### Summary
By default if we have https VS and redirect true ( by default) there is automatically created redirect VS
form port 80 with name-Redirect-

Problem appears if customer have similar VS-es destination address, source address, service port, ip-protocol all match except those VS-es listen on different specified vlans.

Deploy of such AS3 declaration result in similar error:
"message": "Deployment stage 'Deploy AS3 declaration' failed with exception: AS3 declaration deployment error: At least one of the applications has failed to deploy. Results: [{'code': 200, 'message': 'no change', 'host': 'localhost', 'tenant': 'Common', 'runTime': 544}, {'code': 422, 'message': 'declaration failed', 'response': '01070333:3: Virtual Server /tenant_10/application_10/vs_doc_ltm_internal_v4_443-Redirect- illegally shares destination address, source address, service port, ip-protocol, and vlan with Virtual Server /tenant_10/application_10/vs_doc_ltm_v4_443-Redirect-.', 'host': 'localhost', 'tenant': 'tenant_10', 'runTime': 1164}, {'code': 200, 'message': 'no change', 'host': 'localhost', 'tenant': 'Common', 'runTime': 594}]",

Cause of this is that redirect VS is created on all vlans without taking in consideration port 443 VS vlan allowed config
and in fact attempt to create two same -Redirect- VS-es.

### Steps To Reproduce

EXAMPLE #1

1. Submit the following declaration:
```json
{
    "class": "ADC",
    "schemaVersion": "3.23.0",
    "id": "urn:uuid:23a29f70-d6bd-4e2b-b704-d07bbbc77679",
    "label": "Converted Declaration",
    "remark": "Generated by Project Journeys",
    "tenant_12": {
        "class": "Tenant",
        "application_12": {
            "class": "Application",
            "template": "generic",
            "http_xff_unitynet-spre": {
                "class": "HTTP_Profile"
            },
            "pool_v4_443": {
                "members": [
                    {
                        "addressDiscovery": "static",
                        "servicePort": 443,
                        "serverAddresses": [
                            "1.2.2.2",
                            "1.2.2.4"
                        ],
                        "shareNodes": true
                    }
                ],
                "monitors": [
                    {
                        "bigip": "/Common/gateway_icmp"
                    }
                ],
                "class": "Pool"
            },
            "vs_ltm_v4_443": {
                "layer4": "tcp",
                "pool": "pool_v4_443",
                "translateServerAddress": true,
                "translateServerPort": false,
                "class": "Service_HTTPS",
                "serverTLS": "clientssl_v4_443",
                "profileHTTP": {
                    "use": "/tenant_12/application_12/http_xff_unitynet-spre"
                },
                "clientTLS": {
                    "bigip": "/Common/serverssl-insecure-compatible"
                },
                "virtualAddresses": [
                    "10.10.10.10%400"
                ],
                "persistenceMethods": [],
                "snat": "auto",
                "allowVlans": [
                    {
                        "bigip": "/Common/chp_400_external"
                    }
                ]
            },
            "cert_2022_03": {
                "class": "Certificate",
                "certificate": {
                    "bigip": "/Common/default.crt"
                },
                "privateKey": {
                    "bigip": "/Common/default.key"
                }
            },
            "clientssl_v4_443": {
                "certificates": [
                    {
                        "certificate": "/Common/default.crt"
                    }
                ],
                "class": "TLS_Server",
                "insertEmptyFragmentsEnabled": true
            }
        }
    }
}
```

2. Observe the following error response:

Device itself have created two VS-es:
```json
ltm virtual /tenant_12/application_12/vs_ltm_v4_443 {
    creation-time 2021-09-23:07:05:38
    description application_12
    destination /tenant_12/10.10.10.10%400:443
    ip-protocol tcp
    last-modified-time 2021-09-23:07:05:38
    mask 255.255.255.255
    pool /tenant_12/application_12/pool_v4_443
    profiles {
        /Common/f5-tcp-progressive { }
        /Common/serverssl-insecure-compatible {
            context serverside
        }
        /tenant_12/application_12/clientssl_v4_443 {
            context clientside
        }
        /tenant_12/application_12/http_xff_unitynet-spre { }
    }
    serverssl-use-sni disabled
    source 0.0.0.0%400/0
    source-address-translation {
        type automap
    }
    translate-address enabled
    translate-port disabled
    vlans {
        /Common/chp_400_external  <---------------------original have vlan specified
    }
    vlans-enabled
}
ltm virtual /tenant_12/application_12/vs_ltm_v4_443-Redirect- {
    creation-time 2021-09-23:07:05:38
    description application_12
    destination /tenant_12/10.10.10.10%400:80
    ip-protocol tcp
    last-modified-time 2021-09-23:07:05:38
    mask 255.255.255.255
    profiles {
        /Common/f5-tcp-progressive { }
        /Common/http { }
    }
    rules {
        /Common/_sys_https_redirect
    }
    serverssl-use-sni disabled
    source 0.0.0.0%400/0
    source-address-translation {
        type automap
    }
    translate-address enabled
    translate-port enabled
}
  <------------------there is no vlan list here so all vlans allowed.
```

EXAMPLE #2

1. Submit the following declaration:
```json
{
    "class": "ADC",
    "schemaVersion": "3.23.0",
    "id": "urn:uuid:4b86cd48-fde0-4c9e-a5e3-e2fa7aef0988",
    "label": "Converted Declaration",
    "remark": "Generated by Project Journeys",
    "tenant_10": {
        "class": "Tenant",
        "application_10": {
            "class": "Application",
            "template": "generic",
            "vs_doc_ltm_v4_443": {
                "layer4": "tcp",
                "pool": "pool_ows_v4_443",
                "iRules": [
                    {
                        "use": "/tenant_10/application_10/ir_rule_talk-deny"
                    }
                ],
                "translateServerAddress": true,
                "translateServerPort": true,
                "class": "Service_HTTPS",
                "serverTLS": "clientssl_profile_test",
                "profileHTTP": {
                    "use": "/tenant_10/application_10/http_docs_prof"
                },
                "profileHTTPCompression": {
                    "bigip": "/Common/httpcompression"
                },
                "profileTCP": {
                    "bigip": "/Common/tcp"
                },
                "virtualAddresses": [
                    "11.11.11.11%100"
                ],
                "persistenceMethods": [],
                "snat": "none",
                "allowVlans": [
                    {
                        "bigip": "/Common/cip_100_external"
                    }
                ]
            },
            "http_docs_prof": {
                "fallbackRedirect": "https://blalba.com/maintenance/blabla.com/SorryPage.html",
                "insertHeader": {
                    "name": "X-Forwarded-IP",
                    "value": "[expr { [IP::client_addr] }]"
                },
                "class": "HTTP_Profile"
            },
            "pool_ows_v4_443": {
                "members": [
                    {
                        "addressDiscovery": "static",
                        "servicePort": 80,
                        "serverAddresses": [
                            "13.13.13.14",
                            "13.13.13.15"
                        ],
                        "shareNodes": true
                    }
                ],
                "monitors": [
                    {
                        "use": "/Common/Shared/monitor_HTTP"
                    }
                ],
                "class": "Pool"
            },
            "ir_rule_talk-deny": {
                "class": "iRule",
                "iRule": {
                    "base64": "ICAgICAgICAjIERPIE5PVCBESVNBQkxFIHdpdGhvdXQgdGFsa2luZyB0byB0aGUgSVQgQlNQIHRlYW0gKFdpbGxpYW0gU2NhcmNlL05vYWggR2VvcmdlKSBhbmQgb3BlbmluZyBhIENIRyB0aWNrZXQuCiMKIyBXaXRob3V0IHRoaXMgNDA0IGJsb2NrLCB0aGUgc2VydmVycyBjYW4gYmUgb3ZlcndoZWxtZWQgYW5kIGNyYXNoLCBidXQgUEQgbmVlZHMgdGhlIFRhbGsgcGx1Z2luIGluIENvbmZsdWVuY2UKd2hlbiBIVFRQX1JFUVVFU1QgewoJaWYgeyBbSFRUUDo6cGF0aF0gc3RhcnRzX3dpdGggIi9yZXN0L3RhbGsvMS4wL2NvbW1lbnRzL3VwZGF0ZXMiICB9IHsKCQkjY2hlY2tzIGZvciBUQUxLIGNvbW1lbnQgdXBkYXRlcywgZXZlcnkgMjAgc2Vjb25kcwoJCUhUVFA6OnJlc3BvbmQgNDA0CgkJcmV0dXJuCgl9Cn0="
                }
            },
            "cert_2019_09": {
                "class": "Certificate",
                "certificate": {
                    "bigip": "/Common/default.crt"
                },
                "privateKey": {
                    "bigip": "/Common/default.key"
                }
            },
            "clientssl_profile_test": {
                "certificates": [
                    {
                        "certificate": "cert_2019_09"
                    }
                ],
                "class": "TLS_Server",
                "insertEmptyFragmentsEnabled": true
            },
            "vs_doc_ltm_internal_v4_443": {
                "layer4": "tcp",
                "pool": "pool_ows_v4_443",
                "iRules": [
                    {
                        "use": "/tenant_10/application_10/ir_rule_talk-deny"
                    }
                ],
                "translateServerAddress": true,
                "translateServerPort": true,
                "class": "Service_HTTPS",
                "serverTLS": "clientssl_profile_test",
                "profileHTTP": {
                    "bigip": "/Common/http"
                },
                "profileTCP": {
                    "bigip": "/Common/tcp"
                },
                "virtualAddresses": [
                    "11.11.11.11%100"
                ],
                "persistenceMethods": [],
                "snat": {
                    "use": "/tenant_10/application_10/spool_docs_snat"
                },
                "allowVlans": [
                    {
                        "bigip": "/Common/cip_100_internal"
                    }
                ]
            },
            "spool_docs_snat": {
                "snatAddresses": [
                    "13.13.13.20"
                ],
                "class": "SNAT_Pool"
            },
            "ir_redirect": {
                "class": "iRule",
                "iRule": {
                    "base64": "ICAgICAgICAgICAgICAgICN3aGVuIENMSUVOVF9BQ0NFUFRFRCB7CiMJbG9nIGxvY2FsMC4gIkJMQUggPT4gW0lQOjpjbGllbnRfYWRkcl0gPFtUQ1A6OnBheWxvYWQgbGVuZ3RoXT4iCiMJVENQOjpjb2xsZWN0IDEwCiN9CiMKI3doZW4gQ0xJRU5UX0RBVEEgewojCWxvZyBsb2NhbDAuICJCTEFIID0+IFtUQ1A6OnBheWxvYWRdIgojCVRDUDo6cmVsZWFzZQojfQojCiN3aGVuIFNFUlZFUl9DT05ORUNURUQgewojCWxvZyBsb2NhbDAuICJCTEFIID0+IFtJUDo6Y2xpZW50X2FkZHJdIgojfQp3aGVuIEhUVFBfUkVRVUVTVCB7Cglsb2cgbG9jYWwwLiAiUmVkaXJlY3QgUmVxdWVzdGVkIgoJSFRUUDo6cmVzcG9uZCAzMDEgbG9jYXRpb24gImh0dHBzOi8vZG9jcy5mNW5ldC5jb20iCn0="
                }
            }
        }
    },
    "Common": {
        "class": "Tenant",
        "Shared": {
            "class": "Application",
            "template": "shared",
            "monitor_HTTP": {
                "adaptive": false,
                "interval": 25,
                "dscp": 0,
                "receive": "SERVICE STATUS: UP",
                "send": "GET /monitors/fastcheck.cgi?site=cfl HTTP/1.1\\r\\nHost: docs.f5net.com\\r\\nUser-Agent: Big-IP Monitor\\r\\nConnection: Close\\r\\n\\r\\n",
                "timeout": 76,
                "class": "Monitor",
                "monitorType": "http",
                "targetPort": 0
            }
        }
    }
}
```
2. Observe the following error response:
```json
"message": "Deployment stage 'Deploy AS3 declaration' failed with exception: AS3 declaration deployment error: At least one of the applications has failed to deploy. Results: [{'code': 200, 'message': 'no change', 'host': 'localhost', 'tenant': 'Common', 'runTime': 544}, {'code': 422, 'message': 'declaration failed', 'response': '01070333:3: Virtual Server /tenant_10/application_10/vs_doc_ltm_internal_v4_443-Redirect- illegally shares destination address, source address, service port, ip-protocol, and vlan with Virtual Server /tenant_10/application_10/vs_doc_ltm_v4_443-Redirect-.', 'host': 'localhost', 'tenant': 'tenant_10', 'runTime': 1164}, {'code': 200, 'message': 'no change', 'host': 'localhost', 'tenant': 'Common', 'runTime': 594}]",
```

### Expected Behavior
Expected to create redirect VS with same vlan list as original 443 VS

### Actual Behavior
Behavior described on EXAMPLE #1

Related to this feature: https://github.com/f5devcentral/f5-as3-config-converter/issues/12 not to this specific issue.

**Comment by dstokesf5** [comment]
Thank you for your feedback. I have added this issue to our internal product backlog as AUTOTOOL-2782.

**Comment by dstokesf5** [comment]
This has been resolved in the 3.33.0 release.
